lib/utils/cache.js

```diff
@@ -79,8 +90,39 @@ module.exports = class Cache {
         key,
         value: SafeJSON.stringify(value)
       });
-    } catch (err) {
-      return;
-    }
+    } catch (err) { /* passthrough */ }
   }
-};
+
+  /**
+   * Retrieve a value from the cache by key.
+   *
+   * @param   {Object}   opts           Options object.
+   * @param   {Function} opts.action    Raw function to invoke (returns promise).
+   * @param   {Function} opts.hashArgs  Optional hash argument function.
+   * @returns {Function}                Wrapped action with cache get/set.
+   */
+  wrapAction(opts) {
```

Comment:
Really nice to see this abstracted! I think we could bikeshed on the method name, although I'm not sure what a better alternative would be.